libexec/rtld-elf/aarch64/rtld_c18n_asm.S

```diff
@@ -101,43 +100,64 @@ ENTRY(_rtld_get_rstk)
 
 	mov	w0, w14					/* w0 = cid */
 	bl	get_rstk
-	mov	c11, c0
+	mov	c17, c0
```

Comment:
I'm not a huge fan of all the non-standard calling conventions for the various hooks, in particular when there's no documentation about what's going on. Documentation for the standard calling convention isn't needed, because one expects that anything put in [cwx]0 is the return value, but using a random register like c17 to communicate between pieces of code *really* needs some documentation. Including for why c17 was picked (presumably because it's IP1 and can be used as a temporary register between function calls, though why not c16 i.e. IP0?...).

Reply:
c17 is used merely because it is the register allocated to hold the target restricted stack in the trampoline. It could have been any other temporary register. I will document it at the top of `_rtld_get_rstk`, which is only supposed to be called from the trampoline.